src/controller/AutoCommissioner.cpp

```diff
@@ -121,6 +121,12 @@ CommissioningStage AutoCommissioner::GetNextCommissioningStageInternal(Commissio
     case CommissioningStage::kSecurePairing:
         return CommissioningStage::kReadCommissioningInfo;
     case CommissioningStage::kReadCommissioningInfo:
+        if (mDeviceCommissioningInfo.general.breadcrumb > 0)
+        {
+            // We failed on network setup or later, the node failsafe has not been re-armed and the breadcrumb has not been reset.
```

Comment:
It's not clear to me how we know we failed "on network setup or later".  Is that based on the breadcrumb being nonzero?  But my understanding (which could be totally wrong!) is that the breadcrumb is the last stage we completed successfully.  So we would probably want to be comparing to kSendNOC or something around there, not 0....

Or am I completely missing something?

Reply:
Actually, I take that back - the comparison to 0 does need to happen - I've reverted this in a more recent commit. Very few commands actually take and set the breadcrumb. The only one prior to add noc that does so is the failsafe. I suppose I could compare to arm failsafe, but that's even less intuitive. I've added a comment.